wire-ios/Wire-iOS/Sources/UserInterface/UserProfile/Devices/UserClientListViewController.swift

```diff
@@ -165,3 +189,42 @@ extension UserClientListViewController: ParticipantDeviceHeaderViewDelegate {
         URL.wr_fingerprintLearnMore.openInApp(above: self)
     }
 }
+
+extension Array where Element: UserClientType {
+
+    @MainActor
+    func updateCertificates(mlsGroupId: MLSGroupID, userSession: UserSession) async -> [UserClientType] {
```

Comment:
@theswiftway I would also like to improve the readability and testability of this method.

Reply:
probably we can move it outside of the viewController to a viewModel or usecase if suitable

Reply:
I would like rewrite this entire class to make it more testable and migrate it to swiftUI. May be a task post Zulu

Reply:
https://wearezeta.atlassian.net/browse/WPB-6798